modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HTTPServicesRegistry.java

```diff
@@ -248,28 +248,16 @@ private void validateResourceSignature(HttpResource resource) {
             throw new BallerinaConnectorException("resource signature parameter count should be more than two");
         }
 
-        //Validate request parameter
-        ParamDetail reqParamDetail = paramDetails.get(0);
-        if (reqParamDetail == null) {
-            throw new BallerinaConnectorException("request parameter cannot be null");
-        }
-        if (reqParamDetail.getVarType().getPackagePath() == null
-                || !reqParamDetail.getVarType().getPackagePath().equals(Constants.PROTOCOL_PACKAGE_HTTP)
-                || !reqParamDetail.getVarType().getName().equals(Constants.REQUEST)) {
-            throw new BallerinaConnectorException("request parameter should be of type - "
-                                                          + Constants.PROTOCOL_PACKAGE_HTTP + ":" + Constants.REQUEST);
+        //validate connection parameter
+        if (isValidParam(paramDetails.get(0), Constants.CONNECTION)) {
```

Comment:
Shall we remove the comment and change the function name to isValidResourceParam ?

Reply:
Ack